conans/test/functional/toolchains/cmake/test_cmake_toolchain.py

```diff
@@ -115,3 +115,69 @@ def test_cmake_toolchain_without_build_type():
     toolchain = client.load("conan_toolchain.cmake")
     assert "CMAKE_MSVC_RUNTIME_LIBRARY" not in toolchain
     assert "CMAKE_BUILD_TYPE" not in toolchain
+
+
+@pytest.mark.parametrize("with_context", ["False", "True"])
+def test_cmake_toolchain_multiple_user_toolchain(with_context):
+    """ A consumer consuming two packages that declare:
+            self.conf_info["tools.cmake.cmaketoolchain:user_toolchain"]
+        The consumer wants to use apply both toolchains in the CMakeToolchain.
+        There are two ways to customize the CMakeToolchain (parametrized):
+                1. Altering the context of the block (with_context = True)
+                2. Using the t.blocks["user_toolchain"].user_toolchains = [] (with_context = False)
+    """
+    client = TestClient()
+    conanfile = textwrap.dedent("""
+        import os
+        from conans import ConanFile
+        class Pkg(ConanFile):
+            exports_sources = "*"
+            def package(self):
+                self.copy("*")
+            def package_info(self):
+                f = os.path.join(self.package_folder, "mytoolchain.cmake")
+                self.conf_info["tools.cmake.cmaketoolchain:user_toolchain"] = f
+        """)
+    client.save({"conanfile.py": conanfile,
+                 "mytoolchain.cmake": 'message(STATUS "mytoolchain1.cmake !!!running!!!")'})
+    client.run("create . toolchain1/0.1@")
+    client.save({"conanfile.py": conanfile,
+                 "mytoolchain.cmake": 'message(STATUS "mytoolchain2.cmake !!!running!!!")'})
+    client.run("create . toolchain2/0.1@")
+
+    conanfile = textwrap.dedent("""
+        from conans import ConanFile
+        from conan.tools.cmake import CMake, CMakeToolchain
+        class Pkg(ConanFile):
+            settings = "os", "compiler", "arch", "build_type"
+            exports_sources = "CMakeLists.txt"
+            tool_requires = "toolchain1/0.1", "toolchain2/0.1"
+
+
+            def generate(self):
+                # Get the toolchains from "tools.cmake.cmaketoolchain:user_toolchain" conf at the
+                # tool_requires
+                user_toolchains = []
+                for dep in self.dependencies.direct_build.values():
+                    ut = dep.conf_info["tools.cmake.cmaketoolchain:user_toolchain"]
+                    if ut:
+                        user_toolchains.append(ut)
+
+                # Modify the context of the user_toolchain block
+                t = CMakeToolchain(self)
+                if {}:
+                    t.blocks["user_toolchain"].values["paths"] = user_toolchains
```

Comment:
Maybe if the block has a explicit ``.user_toolchains`` attribute, the generic context one doesn't need to be documented (or even tested?)

Reply:
My point is, you can always do that because of the blocks, right? even if it is not documented you can still watch it.
So, Is it worth it to have the `.user_toolchains` or just the context? As I said in the PR description, I don't think we should keep both.

Reply:
Oh, sorry, I missed the updated description of the PR. 
Yes, I agree we should probably do only 1 way to do this. As the context one is very built-in and cannot be removed, maybe we just want to remove the ``.user_toolchains`` attribute. The truth is that those attributes were added before we had the ``block.values`` helper, that greatly simplify to modify contexts.